src/ChannelServer/World/Entities/Components/Buffs.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Aura development team - Licensed under GNU GPL
+// For more information, see licence.txt in the main folder
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Melia.Channel.Network;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Const;
+using Melia.Shared.Util;
+
+namespace Melia.Channel.World
+{
+	/// <summary>
+	/// Character buffs.
+	/// </summary>
+	public class Buffs
+	{
+		private readonly Dictionary<BuffId, Buff> _buffs = new Dictionary<BuffId, Buff>();
+
+		/// <summary>
+		/// The owner of this object.
+		/// </summary>
+		public Character Character { get; }
+
+		/// <summary>
+		/// Creates new instance for character.
+		/// </summary>
+		/// <param name="character"></param>
+		public Buffs(Character character)
+		{
+			this.Character = character ?? throw new ArgumentNullException(nameof(character));
+		}
+
+		/// <summary>
+		/// Returns the amount of buffs in the collection.
+		/// </summary>
+		public int Count { get { lock (_buffs) return _buffs.Count; } }
+
+		/// <summary>
+		/// Adds given without updating the client. Replaces existing
+		/// buffs.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void AddSilent(Buff buff)
+		{
+			lock (_buffs)
+				_buffs[buff.Id] = buff;
+		}
+
+		/// <summary>
+		/// Adds given buff and updates the client. Replaces existing
+		/// buffs.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void Add(Buff buff)
+		{
+			var buffExists = _buffs.ContainsKey(buff.Id);
```

Comment:
You need to lock collections before accessing them, since the collection could change at _any_ point in a multi-threaded environment.

Reply:
Will keep it in mind for accessing collections.